float8_experimental/float8_linear.py

```diff
@@ -335,7 +342,7 @@ def forward(self, x):
         y = self.cast_y_to_float8_in_bw(y, self.emulate)
 
         if self.bias is not None:
-            y = y + self.bias.to(self.bias_dtype)
+            y = y + self.bias.to(y.dtype)
```

Comment:
this change is not called out in the configs, but this just removes the need to store a module attribute, which also makes this code more full-graph compile friendly.   Modifying module attributes seems to graph break if autocast and FSDP are both enabled.